### PR TITLE
Better handling of missing inputs in get_total_irradiance, get_sky_diffuse

### LINKED ISSUE #949: Improve docstring or behavior for irradiance.get_total_irradiance and irradiance.get_sky_diffuse
`pvlib.irradiance.get_total_irradiance` accepts kwargs `dni_extra` and `airmass`, both default to `None`. However, values for these kwargs are required for several of the irradiance transposition models. 

See discussion [here](https://groups.google.com/d/msg/pvlib-python/ZPMdpQOD6F4/cs1t23w8AwAJ)

Docstring should specify when `dni_extra` and `airmass` are required, and which airmass is appropriate for each model.

Could also test for kwarg values if e.g. `model=='perez'`

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #949 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] ~~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Missing `airmass` can be calculated. 

Calculating missing `dni_extra` requires having a datetime index. Since the functions are not constrained to input type Series, I suggest that pvlib raise an informative ValueError and let the user handle it.

### CODE REVIEW

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:362]:
```suggestion
    Models ``'haydavies'``, ``'reindl'``, or ``'perez'`` require ``dni_extra``. Values can
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:365]:
```suggestion
    The ``'perez'`` model requires relative airmass (``airmass``) as input. If
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:366]:
```suggestion
    ``airmass`` is not provided, it is calculated using the defaults in
```
I can't remember are kwargs supposed to use single backticks or double? I think we have both conventions. Single backticks show up as italics, while double show as syntax highlighting.

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:367]:
```suggestion
    :py:func:`~pvlib.atmosphere.get_relative_airmass`.
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:402]:
```suggestion
        Panel tilt from horizontal. [degree]
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:421]:
```suggestion
        Irradiance model. Can be one of ``'isotropic'``, ``'klucher'``, ``'haydavies'``,
        ``'reindl'``, ``'king'``, ``'perez'``.
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:423]:
```suggestion
        Used only if ``model='perez'``. See :py:func:`~pvlib.irradiance.perez`.
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:359]:
```suggestion

    Raises
    ------
    ValueError
        If model is one of ``'haydavies'``, ``'reindl'``, or ``'perez'`` and ``dni_extra``
        is ``None``.

```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:443]:
```suggestion
        If model is one of ``'haydavies'``, ``'reindl'``, or ``'perez'`` and ``dni_extra``
        is ``None``.

    Notes
    -----
    Models ``'haydavies'``, ``'reindl'``, or ``'perez'`` require ``dni_extra``. Values can
    be calculated using :py:func:`~pvlib.irradiance.get_extra_radiation`.

    The ``'perez'`` model requires relative airmass (``airmass``) as input. If
    ``airmass`` is not provided, it is calculated using the defaults in
    :py:func:`~pvlib.atmosphere.get_relative_airmass`.
```

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:553]: nit: we should decide on a convention, is it W/m2 or W/m^2? I can't decide. Let's put it to the group.

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:556]: We can do better than this, pvsyst gives a rough table here: https://www.pvsyst.com/help/albedo.htm then copied verbatim into [SolarFarmer](https://dnvgldocs.azureedge.net/SolarFarmer_Latest/CalcRef/Theory/Transposition.html#ground-albedo-irradiance) and [pvpmc](https://pvpmc.sandia.gov/modeling-steps/1-weather-design-inputs/plane-of-array-poa-irradiance/calculating-poa-irradiance/poa-ground-reflected/albedo/). At the very least we can give snow (0.6-0.8), dirt and grass (0.2-0.25) and water (0.1). I think we should raise a new issue for this though, I wouldn't try to change it here in this PR, it's unrelated. 😉

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:345]:
```suggestion
    surface_type : None or str, default None
```

I think if you use `str` then Sphinx will create a link to the Python docs

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:348]:
```suggestion
    model : str, default 'isotropic'
```
ditto

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:419]:
```suggestion
    model : str, default 'isotropic'
```
ditto

**Comment by mikofski** [comment]
Thanks @cwhanse !

File: pvlib/irradiance.py
Comment (cwhanse) [reply pvlib/irradiance.py:366]: The intent here is to refer to the kwarg, so syntax highlighting (double backticks).

File: pvlib/irradiance.py
Comment (cwhanse) [reply pvlib/irradiance.py:359]: I agree that the `ValueError` is raised, but it is raised in a function that is called by `get_total_irradiance`. So include the `Raises` block here also?

File: pvlib/irradiance.py
Comment (cwhanse) [reply pvlib/irradiance.py:556]: Agree. Edits are creeping outwards e.g. the unit notation in `get_ground_diffuse`. I'll resist the urge.

File: pvlib/tests/test_irradiance.py
Comment (wholmgren) [inline pvlib/tests/test_irradiance.py:300]:
```suggestion
    # test assumes location is Tucson, AZ
```

Comment (wholmgren) [reply pvlib/tests/test_irradiance.py:300]: https://en.wikipedia.org/wiki/Tucson,_Arizona

>The Spanish name of the city, Tucsón [tukˈson], is derived from the O'odham Cuk Ṣon [tʃʊk ʂɔːn], meaning "(at the) base of the black [hill]",

**Review by wholmgren — APPROVED** [review]

File: pvlib/irradiance.py
Comment (mikofski) [reply pvlib/irradiance.py:362]: you +1 this, but it wasn't committed

File: pvlib/irradiance.py
Comment (mikofski) [inline pvlib/irradiance.py:444]:
```suggestion
    :py:func:`~pvlib.atmosphere.get_relative_airmass`.
```
lightning always strikes twice. missed one

File: pvlib/irradiance.py
Comment (mikofski) [reply pvlib/irradiance.py:359]: IDK. @wholmgren ? I think the user doesn't actually know that it calls `get_diffuse_sky()` so either:
* alerting them to why it may raise `ValueError` is polite and not redundant
* or add somewhere that this calls `get_diffuse_sky()` possible in a _See Also_ section?
your call. I will not choose the color of the _white_ bike shed 😉

Comment (wholmgren) [reply pvlib/irradiance.py:359]: In my review this morning I started typing an opinion on this and then deleted it because I changed my mind twice. Not going near that bike shed with a 10 foot roller. Probably no wrong answer here.

Comment (cwhanse) [reply pvlib/irradiance.py:359]: There's paint on the shed - good enough for me.

A future maintainer may wonder why the docstring has a `Raises` clause when the method contains no `raise` statement. That's my objection.

File: pvlib/tests/test_irradiance.py
Comment (cwhanse) [reply pvlib/tests/test_irradiance.py:300]: wonder how many times I've mispelled 'Tucson' in the past....

**Comment by mikofski** [comment]
Thanks Cliff. Can I push the button when the tests are done?

**Comment by cwhanse** [comment]
Push the green button - yes. Not the red one.

**Comment by mikofski** [comment]
🚀

**Pull request merged**
